src/Core/Abstractions/WellKnownDirectives.cs

```diff
@@ -7,5 +7,6 @@ internal static class WellKnownDirectives
         public const string IfArgument = "if";
         public const string Deprecated = "deprecated";
         public const string DeprecationReasonArgument = "reason";
+        public const string DeprecationDefaultReason = "No longer supported";
```

Comment:
Does having a default reason make sense if the specification says there doesn't need to be one?

Reply:
https://graphql.github.io/graphql-spec/June2018/#sec--deprecated

The spec defines there is one. That means whenever the argument is null we take the default reason.


https://graphql.github.io/graphql-spec/June2018/#sec-Coercing-Field-Arguments

Reply:
> If hasValue is not true and defaultValue exists (including null):
   Add an entry to coercedValues named argumentName with the value defaultValue.

So, this means I have to tune it a little more, If null was provided the it really is null:
@deprecated(reason: null)

But if nothing was provided then it is the default reason:
@deprecated

```graphql
directive @deprecated(
  reason: String = "No longer supported"
) on FIELD_DEFINITION | ENUM_VALUE
```

Reply:
> That means whenever the argument is null we take the default reason.

We have to change it to this: That means whenever the argument is not provided we take the default reason.

Reply:
https://github.com/ChilliCream/hotchocolate/issues/878

Reply:
Whoops missed the default value somehow, sorry about that.

Reply:
just checked back with Ivan and it is like the following:

`@deprecated` => default reason value
`@deprecated(reason: null)` => reason becomes null

I will have to rework and review some parts of the argument coercion.